src/sdt.rs

```diff
@@ -77,6 +77,15 @@ impl SdtHeader {
         str::from_utf8(&self.oem_table_id).unwrap()
     }
 
+    #[cfg(test)]
+    pub fn set_right_checksum(&mut self) {
```

Comment:
Does this require the existing checksum field to be 0 to calculate the right number?

Reply:
Hmm, it depends. Basically it goes through all the fields so it will read the checksum at some point. However I'm not sure if it will work if we have a checksum different from 0 at the beginning. Why?